src/overlays/actors/ovl_En_Poh/z_en_poh.c

```diff
@@ -20,111 +53,1095 @@ const ActorInit En_Poh_InitVars = {
+extern AnimationHeader D_06000570;
+extern AnimationHeader D_06000FE4;
+extern AnimationHeader D_060011C4;
+
+extern Gfx D_06004638[];
+
+extern Gfx D_06004498[];
+extern Gfx D_06004530[];
+
+void EnPoh_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnItem00* collectible;
+    EnPoh* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 30.0f);
+    Collider_InitJntSph(globalCtx, &this->colliderSph);
+    Collider_SetJntSph(globalCtx, &this->colliderSph, &this->actor, &sJntSphInit, &this->colliderSphItem);
+    this->colliderSph.list[0].dim.worldSphere.radius = 0;
+    this->colliderSph.list[0].dim.worldSphere.center.x = this->actor.posRot.pos.x;
+    this->colliderSph.list[0].dim.worldSphere.center.y = this->actor.posRot.pos.y;
+    this->colliderSph.list[0].dim.worldSphere.center.z = this->actor.posRot.pos.z;
+    Collider_InitCylinder(globalCtx, &this->colliderCyl);
+    Collider_SetCylinder(globalCtx, &this->colliderCyl, &this->actor, &sCylinderInit);
+    func_80061ED4(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->unk_194 = 0;
+    this->unk_195 = 32;
+    this->visibilityTimer = Math_Rand_S16Offset(700, 300);
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    Lights_PointGlowSetInfo(&this->lightInfo, this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y,
+                            this->actor.initPosRot.pos.z, 255, 255, 255, 0);
+    if (this->actor.params >= 4) {
+        this->actor.params = EN_POH_NORMAL;
+    }
+    if (this->actor.params == EN_POH_RUPEE) {
+        D_80AE1A50++;
+        if (D_80AE1A50 >= 3) {
+            Actor_Kill(&this->actor);
+        } else {
+            collectible = Item_DropCollectible(globalCtx, &this->actor.posRot.pos, 0x4000 | ITEM00_RUPEE_BLUE);
+            if (collectible != NULL) {
+                collectible->actor.speedXZ = 0.0f;
+            }
+        }
+    } else if (this->actor.params == EN_POH_FLAT) {
+        if (Flags_GetSwitch(globalCtx, 0x28) || Flags_GetSwitch(globalCtx, 0x9)) {
+            Actor_Kill(&this->actor);
+        } else {
+            Flags_SetSwitch(globalCtx, 0x28);
+        }
+    } else if (this->actor.params == EN_POH_SHARP) {
+        if (Flags_GetSwitch(globalCtx, 0x29) || Flags_GetSwitch(globalCtx, 0x9)) {
+            Actor_Kill(&this->actor);
+        } else {
+            Flags_SetSwitch(globalCtx, 0x29);
+        }
+    }
+    if (this->actor.params < EN_POH_SHARP) {
+        this->objectIdx = Object_GetIndex(&globalCtx->objectCtx, OBJECT_POH);
+        this->isComposer = false;
+        this->actor.naviEnemyId = 0x44;
+    } else {
+        this->objectIdx = Object_GetIndex(&globalCtx->objectCtx, OBJECT_PO_COMPOSER);
+        this->isComposer = true;
+        this->actor.naviEnemyId = 0x43;
+    }
+    this->info = &sPoeInfo[this->isComposer];
```

Comment:
`isComposer` is definitely not a boolean, based on this and the == 1 later in the file. It's probably some form of type variable, associated with an enum of values 0 and 1.